docs/madr/decisions/032-add-inbound-in-runtime.md

```diff
@@ -0,0 +1,142 @@
+# Adding inbounds in runtime / Changing Kube Service for the Pod 
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/5086
+
+## Context and Problem Statement
+
+Changing Kubernetes Service for a Pod results in changing / adding a new inbound in Dataplane object.
+
+Adding a new inbound for a Dataplane object results in two parallel operations:
+* Updating changed data plane proxy with new certificate
+* Updating all clients with new endpoint targeting changed data plane proxy
+
+If second applies before the first, it can result in broken traffic.
+
+This can happen with ArgoCD and Blue Green rollout.
+First the pod is selected by `service-preview` and if it passes the analysis the pod is targeted by the `service-active`.
+
+Here is an example of `redis` from `kuma-demo` going through blue-green promotion 
+```yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis
+  namespace: kuma-demo
+spec:
+  ports:
+    - port: 6379
+      protocol: TCP
+      targetPort: 6379
+  selector:
+    app: redis
+    rollouts-pod-template-hash: 778dbdddff
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-preview
+  namespace: kuma-demo
+spec:
+  ports:
+    - port: 6379
+      protocol: TCP
+      targetPort: 6379
+  selector:
+    app: redis
+    rollouts-pod-template-hash: 646998df5c
+```
+Once it's finished, `redis` will change selector to `rollouts-pod-template-hash: 646998df5c` and `redis-preview` will keep `rollouts-pod-template-hash: 646998df5c` selector. 
+
+## Decision Drivers
+
+* Enable users to use ArgoCD blue-green rollout without traffic interruption
+
+## Considered Options
+
+* Wait for certs using Dataplane Insights
+* Wait for certs using XDS callbacks
+* Ignore labels in selector to add ignored listener
+
+## Decision Outcome
+
+Chosen option: "Ignore labels in selector to add ignored listener". While it's the hardest to explain. It has no impact on the perf.
+
+## Pros and Cons of the Options
+
+To consider options, we need to first introduce new concept of "Ignored" listener.
+Instead of `healthy: false/true` we also need another state of the listener.
```

Comment:
1. [I think it's `health.ready: bool`](https://github.com/kumahq/kuma/blob/85541e420968cd549a343f9a52e82346b111001c/api/mesh/v1alpha1/dataplane.proto#L100-L117) not `healthy: bool`.
2. Could you add a sentence with clarification if it means we would replace `health.ready: bool` with `state: Ready | NotReady | Ignored`?